### ISSUE TITLE
`jnp.arctan` silently returns wrong values on CPU for arrays larger than ~32-50 elements (regression in jaxlib 0.10.2, CPU-only)

### ISSUE THREAD

**Issue by AreWeDreaming** [jax-ml/jax#39352]
### Description

**Note**: A lot of the heavy lifting was done by Claude but I manually verified everything stated in this issue.

**Description**

On the CPU backend, `jax.numpy.arctan` returns incorrect values once the input array reaches 50 elements — the cutoff is exact: 49-element arrays are correct, 50-element arrays are not. There is no exception, no NaN, no warning. The wrong values are not a single consistent pattern — at n=50 we see a mix of exact `0.0` (22/47 of the bad elements), the *unmodified input* passed through untouched (19/47), and other values that are neither zero, the input, nor the correct answer (6/47); at larger sizes (64/100/200) we see no exact zeros, but a mix of passthrough and other plausible-looking-but-wrong values. This mixed, position-dependent signature is more consistent with a vectorized-kernel masking/uninitialized-memory bug than a simple formula error. It reproduces identically in eager execution, under `jax.jit`, and inside `jax.disable_jit()` (same mismatch counts in all three modes, deterministic across 15+ repeated runs). It is not present on the GPU (CUDA) backend, and not present in jaxlib 0.10.1 — bisected by us to exactly the 0.10.1 → 0.10.2 upgrade.

**Reproduction**

```python
import jax, jax.numpy as jnp, numpy as np

x1 = np.linspace(-3, 3, 49, dtype=np.float32) + 1e-3
x2 = np.linspace(-3, 3, 50, dtype=np.float32) + 1e-3
z1 = x1**2
z2 = x2**2

out = np.array(jax.jit(jnp.arctan)(z1))
ref = np.arctan(z1)
bad = np.abs(out - ref) > 1e-4
print(f"mismatches: {bad.sum()}/{len(z1)}")

out = np.array(jax.jit(jnp.arctan)(z2))
ref = np.arctan(z2)
bad = np.abs(out - ref) > 1e-4
print(f"mismatches: {bad.sum()}/{len(z2)}")

i = np.argmax(bad)
print(f"z2[{i}] = {z2[i]!r}, jax.jit(arctan) = {out[i]!r}, np.arctan (ref) = {ref[i]!r}")
```

**Actual output (jaxlib 0.10.2, CPU)**

```
n=49 mismatches: 0/49
n=50 mismatches: 47/50
z2[0] = np.float32(8.994001), jax.jit(arctan) = np.float32(0.0), np.arctan (ref) = np.float32(1.460066)
```

**Expected output**

`jax.jit(jnp.arctan)(z2)` should match `np.arctan(z2)` to float32 precision (0 mismatches), as it does on jaxlib 0.10.1 and on the CUDA backend.

**What we've verified**

- **Regression window**: confirmed by direct downgrade — jaxlib 0.10.1 passes (0/50 mismatches at n=50), jaxlib 0.10.2 fails (47/50), same machine, same script.
- **CPU-only**: jaxlib 0.10.2 with the CUDA backend shows 0/50 mismatches at n=50 (and 0 at every other size we tried).
- **Not jit-specific**: eager `jnp.arctan(z)`, `jax.jit(jnp.arctan)(z)`, and `jax.jit(jnp.arctan)(z)` under `jax.disable_jit()` all give identical (wrong) mismatch counts per array size on CPU — this is a backend/codegen issue, not a tracing artifact.
- **Size-dependent, with an exact cutoff at 50**: n=8/16/32/49 are all correct (0 mismatches); n=50 and above all show mismatches (47/50 on a dedicated n=50 array; 56/64, 80/100, 103/128, 103/150, 167/200 when sweeping slices of a larger array) — consistent with a vectorized (SIMD) codegen bug tied to a specific chunk/tile width rather than a scalar math error.
- **Not every transcendental function is affected**: `jnp.sin`, `jnp.tan`, `jnp.log1p` show no mismatches on the same input/sizes; `jnp.exp` differs only at float32 precision noise (~1e-7 relative, expected).
- **Reproduces in both float32 and float64** (`jax_enable_x64=True`).

** How we discovered it **
One of our regression tests in our closed-source repo started failing when upgrading to a newer jax version. This failure appeared locally and on github CLI so it is unlikely that this is specific to my system architecture.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:      0.10.2
jaxlib:   0.10.2
numpy:    2.5.1
python:   3.12.13 (conda-forge, GCC 14.3.0)
platform: Linux-6.18.33.2-microsoft-standard-WSL2-x86_64
CPU:      Intel(R) Core(TM) Ultra 7 265H (AVX2, AVX_VNNI, FMA; no AVX-512)
CPU backend: reproduces (47/50 mismatches)
CUDA backend: does not reproduce (0/50 mismatches), same jax/jaxlib version
```

**Comment by hawkinsp** [comment]
@seantalts @penpornk PTAL?

**Comment by adivijaykumar** [comment]
I also ran into this issue. It seems to stem from the conda-forge build of jax (same versions as noted above), and didn't seem to be present when I installed jax directly through pip.
